astropy/modeling/functional_models.py

```diff
@@ -1719,14 +1723,26 @@ def __init__(
         amplitude_L=amplitude_L.default,
         fwhm_L=fwhm_L.default,
         fwhm_G=fwhm_G.default,
-        method="humlicek2",
+        method=None,
         **kwargs,
     ):
+        if method is None:
+            if HAS_SCIPY:
+                method = "wofz"
+            else:
+                method = "humlicek2"
+
         if str(method).lower() in ("wofz", "scipy"):
             from scipy.special import wofz
 
             self._faddeeva = wofz
         elif str(method).lower() == "humlicek2":
+            warnings.warn(
+                f"{method} has been deprecated since Astropy 5.3 and will be removed in a future version.\n"
+                "It is recommended to always use the `wofz` method which requires `scipy`.",
+                AstropyDeprecationWarning,
+            )
+
             self._faddeeva = self._hum2zpf16c
```

Comment:
```suggestion
        if str(method).lower() == "humlicek2" and HAS_SCIPY:
            warnings.warn(
                f"{method} has been deprecated since Astropy 5.3 and will be removed in a future version.\n"
                "It is recommended to always use the `~scipy.special.wofz` implementation "
                "when `scipy` is installed.",
                AstropyDeprecationWarning,
            )

        if method is None:
            if HAS_SCIPY:
                method = "wofz"
            else:
                method = "humlicek2"

        if str(method).lower() in ("wofz", "scipy"):
            from scipy.special import wofz

            self._faddeeva = wofz
        elif str(method).lower() == "humlicek2":
            self._faddeeva = self._hum2zpf16c
```
I don't have a good feeling about raising a warning when still using the default/only available method, as there is no clear timeline in sight when the non-scipy method might be removed eventually (not sure in fact if it can/should be officially deprecated in 5.3 then).
If there is a test explicitly checking _that_ the warning is raised, that would of course need to be adapted, but I could not find one.

Reply:
> a single warning is due on the initial import of Astropy

👎 on import time warning.

Reply:
I.e. warnings on use in every module that would be affected?

Reply:
I am not following this closely but warning on import seems excessive. Only give warning when it is actually needed, otherwise people going to ignore it, and it is going to cause pain in CI that turns warnings into exceptions.

Reply:
The scenario is, should there be a decision to make SciPy a required dep, say as of release 6.0, should people get a warning that they will not be able to use Astropy in the future if SciPy cannot be imported?
But let's wait for https://github.com/astropy/astropy/issues/12448#issuecomment-1481598597 for follow-up.

Reply:
> should people get a warning that they will not be able to use Astropy in the future if SciPy cannot be imported?

I don't think so? We added things like pyerfa to required dependencies without much fanfare.